lib/publish.js

```diff
@@ -23,8 +23,18 @@ module.exports = async (pluginConfig, {branch, repositoryUrl}, {version, gitHead
   debug('release branch: %o', branch);
   const ref = `refs/tags/${gitTag}`;
 
-  debug('Create git tag %o with commit %o', ref, gitHead);
-  await github.gitdata.createReference({owner, repo, ref, sha: gitHead});
+  try {
+    // Test if the tag already exists
+    await github.gitdata.getReference({owner, repo, ref: `tags/${gitTag}`});
+  } catch (err) {
+    // If the error is 404, the tag doesn't exist, otherwise it's an error
+    if (err.code !== 404) {
+      throw err;
+    }
+    debug('Create git tag %o with commit %o', ref, gitHead);
+    await github.gitdata.createReference({owner, repo, ref, sha: gitHead});
+  }
```

Comment:
I think I would try to create the reference first and then ignore the conflict error if it already exists. That way you only need on request, instead of two

Reply:
I tried that but the problem is there is no indication in the error message that the cause of the error is the tag already existng.

The error message doesn't contain any parsable code. Only a text message. Github might change the phrasing of the error anytime, so it's not safe to rely on that.

And if we receive an error for another reason, we want to throw it and stop the process.

Reply:
got it! In that case, 🚢